xdesign/plot.py

```diff
@@ -361,32 +376,65 @@ def discrete_phantom(phantom, size, ratio=9, uniform=True,
     ValueError
         If size is less than or equal to 0
     """
-    if size <= 0:
+    if psize <= 0:
         raise ValueError('size must be greater than 0.')
 
-    image = 0
+    bounding_box = np.asarray(bounding_box)
+    if bounding_box.min() < 0 or bounding_box.max() > 1:
+        raise ValueError('bounding box must be within (0, 1).')
 
-    if phantom.geometry is not None and phantom.material is not None \
-       and hasattr(phantom.material, prop):
+    size = (bounding_box[1] - bounding_box[0]) / psize
+    size = size.astype('int')
 
-        psize = 1.0 / size
+    image = ret = patch = 0
 
+    prop = np.atleast_1d(prop)
+    if phantom.geometry is not None and phantom.material is not None \
+       and np.alltrue([hasattr(phantom.material, temp) for temp in prop]):
         # Rasterize all geometry in the phantom.
         pmin, patch = discrete_geometry(phantom.geometry, psize, ratio)
 
         # Get the property value
-        value = getattr(phantom.material, prop)(DEFAULT_ENERGY)
-
-        # Make a grid to put store all of the discrete geometries
-        image = np.zeros([size] * phantom.geometry.dim, dtype=float)
+        n_prop = prop.size
+        ret = [None] * n_prop
```

Comment:
I've implemented this change here: mdw771/xdesign#3 because I think that slicing a single array is more compact than slicing each element of a list separately.